### PR TITLE
Updating 4 entities to use the new ui_search approach

### PR DESCRIPTION
**Pull request by kbidarkar**

### CODE REVIEW

**Comment by kbidarkar** [comment]
All the tests pass for 
a) environment and
b) hardware-model

```
(robottelo)[kbidarka@xyz robottelo]$ python -m unittest tests.foreman.ui.test_environment
.....
----------------------------------------------------------------------
Ran 5 tests in 409.727s

OK

(robottelo)[kbidarka@xyz robottelo]$ python -m unittest tests.foreman.ui.test_hw_model
....ss
----------------------------------------------------------------------
Ran 4 tests in 333.987s

OK (skipped=2)
```

**Comment by kbidarkar** [comment]
a) Discovered Hosts 
b) and Hosts 

The above a) and b) needs to be revisited and search results cannot be provided at this stage.

File: robottelo/ui/discoveredhosts.py [hunk 1]
```diff
@@ -8,11 +8,13 @@
 class DiscoveredHosts(Base):
     """Manipulates Discovered Hosts from UI"""
 
-    def search(self, host_name):
-        """Searches existing discovered hosts from UI"""
+    def navigate_to_entity(self):
+        """Navigate to Discovered Hosts entity page"""
         Navigator(self.browser).go_to_discovered_hosts()
-        strategy, value = locators['discoveredhosts.hostname']
-        return self.wait_until_element((strategy, value % host_name))
```
Comment (oshtaier) [inline robottelo/ui/discoveredhosts.py]: Hmm. How did that search work at all? There is no call to `search_entity()` method. We need one test run to make sure that our logic will work here

File: robottelo/ui/hosts.py [hunk 2]
```diff
@@ -198,12 +199,19 @@ def search(self, name):
         """Searches existing host from UI."""
         return self.search_entity(name, locators['host.select_name'])
```
Comment (oshtaier) [inline robottelo/ui/hosts.py]: ???

**Comment by oshtaier** [comment]
Pending comments and at least 1 test per each module (or at least 1 test run with any possible workaround)

**Comment by kbidarkar** [comment]
I am not aware of any workaround for it, apart from using a hardcoded "host" entry in code and testing it out. Let me try it.

**Comment by oshtaier** [comment]
Sure and in that case you will definitely catch exception for `search_entity` call :)

**Comment by kbidarkar** [comment]
```
(robottelo)[kbidarka@xyz robottelo]$ nosetests -m test_search_host tests.foreman.ui.test_host
.
----------------------------------------------------------------------
Ran 1 test in 36.935s

OK
```

The code used to test.

```
class HostSearch(UITestCase, Base):

    @run_only_on('sat')
    def test_search_host(self):
        """@Test: Create a new Host

        @Feature: Host - Positive create

        @Assert: Host can be searched

        """
        with Session(self.browser) as session:
            session.nav.go_to_select_org('Default Organization')
            self.navigator.go_to_hosts()
            search = self.hosts.search('qe-sat6-rhel71.satellite.lab.eng.rdu2.redhat.com')
            self.assertIsNotNone(search)   
```

**Comment by kbidarkar** [comment]
```
(robottelo)[kbidarka@xyz robottelo]$ nosetests -m test_search_discoveryhosts tests.foreman.ui.test_discoveredhosts 
.
----------------------------------------------------------------------
Ran 1 test in 24.979s

OK
```

The code used to test:
```
class DiscoverySearch(UITestCase):
    def test_search_discoveryhosts(self):
        """@Test: Search a Discovered Host

        @Feature: Foreman Discovery

        @Assert: Searching of discovered Host should be successful.

        """
        with Session(self.browser) as session:
            session.nav.go_to_select_org("Default Organization")
            self.assertIsNotNone(self.discoveredhosts.search('mac525400632930'))
```

**Comment by kbidarkar** [comment]
```
(robottelo)[kbidarka@xyz robottelo]$ nosetests -m test_delete_discoveryhosts tests.foreman.ui.test_discoveredhosts
.
----------------------------------------------------------------------
Ran 1 test in 32.017s

OK
```
The code used to test:

```
class DiscoverySearch(UITestCase):
    def test_delete_discoveryhosts(self):
        """@Test: Delete a Discovered Host

        @Feature: Foreman Discovery

        @Assert: Deleting of discovered Host should be successful.

        """
        with Session(self.browser) as session:
            session.nav.go_to_select_org("Default Organization")
            self.assertIsNone(self.discoveredhosts.delete('mac5254004b9f48'))
```

File: robottelo/ui/discoveredhosts.py (hunk 1 above)
Comment (kbidarkar) [inline robottelo/ui/discoveredhosts.py]: Provided the test results below. IT was just that earlier the approach was different, but it is very much compatible with our base/search function.

**Comment by oshtaier** [comment]
ACK

**Comment by abalakh** [comment]
ACK

**Pull request merged**